flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/matrix/Vector.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.matrix;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.Serializable;
+
+/**
+ * The Vector class defines some common methods for both DenseVector and
+ * SparseVector.
+ */
+public abstract class Vector implements Serializable {
+
+	/**
+	 * Parse a DenseVector from a formatted string.
+	 */
+	public static DenseVector dense(String str) {
+		return DenseVector.deserialize(str);
+	}
+
+	/**
+	 * Parse a SparseVector from a formatted string.
+	 */
+	public static SparseVector sparse(String str) {
+		return SparseVector.deserialize(str);
+	}
+
+	/**
+	 * To check whether the formatted string represents a SparseVector.
+	 */
+	public static boolean isSparse(String str) {
+		if (org.apache.flink.util.StringUtils.isNullOrWhitespaceOnly(str)) {
+			return true;
+		}
+		return StringUtils.indexOf(str, ':') != -1 || StringUtils.indexOf(str, "$") != -1;
+	}
+
+	/**
+	 * Parse the tensor from a formatted string.
+	 */
+	public static Vector deserialize(String str) {
```

Comment:
Since there are only two sub classes (i.e., DenseVector and SparseVector), and there should  probably be no more sub classes. So I guess the problem is not that bad when considering the convenience it brings to algorithm developers.